modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/instance/lifecycle/AccountEntryGroupSettingsPortalInstanceLifecycleListener.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.internal.instance.lifecycle;
+
+import com.liferay.account.settings.AccountEntryGroupSettings;
+import com.liferay.commerce.configuration.CommerceAccountGroupServiceConfiguration;
+import com.liferay.commerce.constants.CommerceConstants;
+import com.liferay.commerce.product.model.CommerceChannel;
+import com.liferay.commerce.product.service.CommerceChannelLocalService;
+import com.liferay.commerce.util.AccountEntryAllowedTypesUtil;
+import com.liferay.portal.instance.lifecycle.BasePortalInstanceLifecycleListener;
+import com.liferay.portal.instance.lifecycle.PortalInstanceLifecycleListener;
+import com.liferay.portal.kernel.module.configuration.ConfigurationProvider;
+import com.liferay.portal.kernel.service.CompanyLocalService;
+import com.liferay.portal.kernel.settings.GroupServiceSettingsLocator;
+import com.liferay.portal.kernel.util.LoggingTimer;
+
+import java.util.List;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Danny Situ
+ */
+@Component(service = PortalInstanceLifecycleListener.class)
+public class AccountEntryGroupSettingsPortalInstanceLifecycleListener
```

Comment:
Can you explain me this class? I see you are setting available types based on the feature flag at company activation (startup).
What if then I enable the feature flag? Will I ble able to use the new Account type?
Thanks

Reply:
Hey @luca-pellizzon, The purpose of this class is to update existing commerceChannels accountEntryTypes. This only effects the store fronts account management, as we restrict it to only show business accounts. The control panel account management will always so all possible account types. it would require a portal restart if the feature flag is changed. Hmm maybe I can create a model listener instead.

Reply:
I see, but then wouldn't be better to be on CommerceChannel instead?

Reply:
@luca-pellizzon actually I don't think I can create a model listener. There is already a update for commerceChannels accountEntryTypes in place, so I don't think we need to worry about that scenario.

It also looks like toggling the feature flag from UI does not work, I think for a dev feature flag it is meant to be set in portal.properties.

In terms of AccountEntryGroupSettingsPortalInstanceLifecycleListener, this listener is just meant to update all channels on restart when the feature flag is toggled, it would either add or remove supplier account type to allowed account types for all existing channels.

Thinking about it, it probably makes more sense to just create a upgrade process to update all existing channels, and not worry about weather the feature flag is enabled or not. We would still hide instances of supplier account type to the end user with the feature flag but the database will have that type regardless if the feature is enabled or not. I think I will proceed with this solution, let me know what you think. Thanks.

Reply:
> it probably makes more sense to just create a upgrade process to update all existing channels
go ahead with that